lib/rules/unit-allowed-list/__tests__/index.js

```diff
@@ -340,3 +347,34 @@ testRule({
 		},
 	],
 });
+
+testRule({
+	ruleName,
+
+	config: [['px', 'em'], { ignoreFunctions: ['hsl', '/^rgb/'] }],
```

Comment:
[suggest] What do you think about adding a test case for a raw `RegExp` object?

E.g.

```suggestion
	config: [['px', 'em'], { ignoreFunctions: ['hsl', '/^rgb/', /-gradient$/] }],
```

Reply:
@ybiquitous regex test has been added